lib/inspect_addedfiles.c

```diff
@@ -127,66 +127,68 @@ static bool addedfiles_driver(struct rpminspect *ri, rpmfile_entry_t *file)
+                subpath = entry->data;
+                localpath = file->localpath;
+
+                /* ensure the paths do not start with '/' */
+                if (*subpath == '/') {
+                    while (*subpath == '/') {
+                        subpath++;
+                    }
                 }
-            }
 
-            if (*localpath == '/') {
-                while (*localpath == '/') {
-                    localpath++;
+                if (*localpath == '/') {
+                    while (*localpath == '/') {
+                        localpath++;
+                    }
                 }
-            }
 
-            if (strprefix(localpath, subpath)) {
-                xasprintf(&params.msg, _("Packages should not contain files or directories starting with `%s` on %s in %s: %s"), entry->data, arch, name, file->localpath);
-                params.noun = _("invalid directory ${FILE} on ${ARCH}");
-                add_result(ri, &params);
-                result = !(params.severity >= RESULT_VERIFY);
-                reported = true;
-                goto done;
+                if (strprefix(localpath, subpath)) {
+                    xasprintf(&params.msg, _("Packages should not contain files or directories starting with `%s` on %s in %s: %s"), entry->data, arch, name, file->localpath);
+                    params.noun = _("invalid directory ${FILE} on ${ARCH}");
+                    add_result(ri, &params);
+                    result = !(params.severity >= RESULT_VERIFY);
+                    reported = true;
+                    goto done;
+                }
             }
         }
-    }
 
-    /* Check for any forbidden path suffixes */
-    if (ri->forbidden_path_suffixes && (ri->tests & INSPECT_ADDEDFILES)) {
-        TAILQ_FOREACH(entry, ri->forbidden_path_suffixes, items) {
-            if (strsuffix(file->localpath, entry->data)) {
-                xasprintf(&params.msg, _("Packages should not contain files or directories ending with `%s` on %s in %s: %s"), entry->data, arch, name, file->localpath);
-                params.noun = _("invalid directory ${FILE} on ${ARCH}");
-                add_result(ri, &params);
-                result = !(params.severity >= RESULT_VERIFY);
-                reported = true;
-                goto done;
+        /* Check for any forbidden path suffixes */
+        if (ri->forbidden_path_suffixes && (ri->tests & INSPECT_ADDEDFILES)) {
+            TAILQ_FOREACH(entry, ri->forbidden_path_suffixes, items) {
+                if (strsuffix(file->localpath, entry->data)) {
+                    xasprintf(&params.msg, _("Packages should not contain files or directories ending with `%s` on %s in %s: %s"), entry->data, arch, name, file->localpath);
+                    params.noun = _("invalid directory ${FILE} on ${ARCH}");
+                    add_result(ri, &params);
+                    result = !(params.severity >= RESULT_VERIFY);
+                    reported = true;
```

Comment:
Same question here, seems you might want to add `free(head);` and `free(tail);` before the `goto`.

Reply:
Same, removed head and tail since they are unused now.